File: avalon-support/avalon-oauth2-support/avalon-oauth2-common/src/test/java/io/github/lishangbu/avalon/oauth2/common/autoconfiguration/PasswordEncoderAutoConfigurationTest.java
package io.github.lishangbu.avalon.oauth2.common.autoconfiguration;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;
import org.springframework.security.crypto.password.PasswordEncoder;

class PasswordEncoderAutoConfigurationTest {

    @Test
    void createsDelegatingPasswordEncoder() {
        PasswordEncoderAutoConfiguration configuration = new PasswordEncoderAutoConfiguration();

        PasswordEncoder encoder = configuration.passwordEncoder();

        assertNotNull(encoder);
        String encoded = encoder.encode("secret");
        assertTrue(encoder.matches("secret", encoded));
    }
}

File: avalon-support/avalon-oauth2-support/avalon-oauth2-common/src/test/kotlin/io/github/lishangbu/avalon/oauth2/common/autoconfiguration/PasswordEncoderAutoConfigurationTest.kt
package io.github.lishangbu.avalon.oauth2.common.autoconfiguration

import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class PasswordEncoderAutoConfigurationTest {
    @Test
    fun createsDelegatingPasswordEncoder() {
        val configuration = PasswordEncoderAutoConfiguration()

        val encoder = configuration.passwordEncoder()

        assertNotNull(encoder)
        val encoded = encoder.encode("secret")
        assertTrue(encoder.matches("secret", encoded))
    }
}
